src/common.js

```diff
@@ -7,6 +7,16 @@ export { envGlobal as global };
 
 export let baseUrl;
 
+export function setBaseUrl(url) {
+  // Verify it's a valid URL
+  const URL = require('url').URL;
+  baseUrl = new URL(url).href;
+}
+
+export function resetBaseUrl() {
+  baseUrl = null;
+}
```

Comment:
I think `baseUrl` should always be set to something - so I would prefer we don't use this function and instead just call `setBaseUrl` again with something.

Reply:
This is for tests only - gets tree shaken out of all builds. In tests, I wanted to be able to test the base url behavior without relying on test execution order.

Reply:
I can remove it in favor of manually resetting the base url

Reply:
The thing is all code paths should assume the baseUrl is always defined I think? Or are all accesses gated?

Reply:
Yes all code paths assume it is defined. But in nodejs, it is not defined until you forcibly set it. I saw an infinite loop (in `getMatch()`) and other issues when baseUrl wasn't defined before import map resolution occurred.